src/opm/input/eclipse/Schedule/KeywordHandlers.cpp

```diff
@@ -2020,6 +2021,23 @@ Well{0} entered with disallowed 'FIELD' parent group:
         sched_state.pavg.update(std::move(wpave));
     }
 
+    void Schedule::handleWVFPEXP(HandlerContext& handlerContext) {
+        for (const auto& record : handlerContext.keyword) {
+            const std::string& wellNamePattern = record.getItem("WELL").getTrimmedString(0);
+            const auto well_names = wellNames(wellNamePattern, handlerContext.currentStep);
+            if (well_names.empty())
+                this->invalidNamePattern(wellNamePattern, handlerContext);
+
+            for (const auto& well_name : well_names) {
+                auto well = this->snapshots.back().wells.get(well_name);
+                auto wvfpexp = std::make_shared<WVFPEXP>(well.getWVFPEXP());
+                wvfpexp->update( record );
+                if (well.updateWVFPEXP(wvfpexp))
```

Comment:
Tiny optimisation here: You should write this as
```C++
if (well.updateWVFPEXP(std::move(wvfpexp)))
```
instead since we're not going to need the shared `wvfpexp` object after this statement.